src/app/util/attribute-storage-helpers.h

```diff
@@ -0,0 +1,75 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * Helpers for manipulating values that are going into and out of the attribute
+ * store.
+ */
+
+#pragma once
+
+#include <lib/support/TypeTraits.h>
+
+#include <limits>
+
+namespace chip {
+namespace app {
+
+template <typename T>
+struct NumericAttributeTraits
+{
+    // StorageType is the type used to represent this C++ type in the attribute
+    // store.
+    using StorageType = T;
+
+    // The value reserved in the value space of StorageType to represent null,
+    // for cases when we have a nullable value.
+    static constexpr StorageType kNullValue =
```

Comment:
is this spec-defined or a choice the SDK makes internally? A comment explaining one way or the other may help.

At first look, I was wondering if it would be simpler to use `::max()` all the time since tthat seems to work for both signed and unsigned values.

Reply:
It's spec-defined in that the spec reserves these values as not being valid values so that implementations can use them to represent null.  I'll add a comment.